test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -221,6 +225,10 @@ void doBlockchainTestNoLog(json_spirit::mValue& _v, bool _fillin)
 		}
 	}
 
+	//Delete source test from the json
+	for (auto i: erase_list)
+		_v.get_obj().erase(i);
```

Comment:
Yes, you are right. I was reading it wrong. Only the erased iterator is invalidated.